http2_general/test_h2_hpack.py

```diff
@@ -840,6 +842,78 @@ def test_invalid_data(self):
         self.assertTrue(client.wait_for_connection_close())
 
 
+class TestHpackTableSizeEncodedInInvalidPlace(TestHpackBase):
+    @parameterize.expand(
+        [
+            param(name="begin", expected_status_code="200"),
+            param(name="middle", expected_status_code="400"),
+            param(name="end", expected_status_code="400"),
+        ]
+    )
+    def test(self, name, expected_status_code):
+        """
+        A change in the maximum size of the dynamic table is signaled
+        via a dynamic table size update (see Section 6.3). This dynamic
+        table size update MUST occur at the beginning of the first
+        header block following the change to the dynamic table size.
+        In HTTP/2, this follows a settings acknowledgment (see Section
+        6.5.3 of [HTTP2]).
+        In this test we send maximum size of the dynamic table in the
+        middle and at the end of the of the first header block.
+        """
+        self.start_all_services()
+
+        table_size = 10
+        size_bytes = encode_integer(table_size, 5)
+        size_bytes[0] |= 0x20
+
+        client = self.get_client("deproxy")
+        client.update_initial_settings()
+        client.send_bytes(client.h2_connection.data_to_send())
+        client.h2_connection.clear_outbound_data_buffer()
+        self.assertTrue(
+            client.wait_for_ack_settings(),
+            "Tempesta foes not returns SETTINGS frame with ACK flag.",
+        )
+
+        stream = client.h2_connection._get_or_create_stream(
+            1, AllowedStreamIDs(client.h2_connection.config.client_side)
+        )
+        stream.state_machine.process_input(StreamInputs.SEND_HEADERS)
+
+        frame = None
+        if name == "begin":
+            frame = HeadersFrame(
+                stream_id=1,
+                data=bytes(size_bytes) + HuffmanEncoder().encode(H2Base.post_request),
+                flags=["END_HEADERS", "END_STREAM"],
+            )
+        elif name == "middle":
+            frame = HeadersFrame(
+                stream_id=1,
+                data=HuffmanEncoder().encode([(":authority", "example.com"), (":path", "/")])
+                + bytes(size_bytes)
+                + HuffmanEncoder().encode([(":scheme", "https"), (":method", "POST")]),
+                flags=["END_HEADERS", "END_STREAM"],
+            )
+        elif name == "end":
+            frame = HeadersFrame(
+                stream_id=1,
+                data=HuffmanEncoder().encode(H2Base.post_request) + bytes(size_bytes),
+                flags=["END_HEADERS", "END_STREAM"],
+            )
```

Comment:
I think that you should use new parameter here.
```python3
param(name="begin", data=(bytes(size_bytes) + HuffmanEncoder().encode(H2Base.post_request)), expected_status_code="200"),
```
and 
```suggestion
        frame = HeadersFrame(
            stream_id=1,
            data=data,
            flags=["END_HEADERS", "END_STREAM"],
        )
```